locator/load_sketch.hh

```diff
@@ -231,36 +329,65 @@ public:
     // Returns the difference in tablet count between highest-loaded shard and lowest-loaded shard.
     // Returns 0 when shards are perfectly balanced.
     // Returns 1 when shards are imbalanced, but it's not possible to balance them.
-    load_type get_shard_imbalance(host_id node) const {
-        auto minmax = get_shard_minmax(node);
-        return minmax.max() - minmax.max();
+    size_t get_shard_tablet_count_imbalance(host_id node) const {
+        auto minmax = get_shard_minmax_tablet_count(node);
+        return minmax.max() - minmax.min();
     }
 
     min_max_tracker<load_type> get_shard_minmax(host_id node) const {
         min_max_tracker<load_type> minmax;
         if (_nodes.contains(node)) {
+            throw_on_incomplete_data(node);
             auto& n = _nodes.at(node);
-            for (auto&& load: n._shards) {
-                minmax.update(load);
+            for (auto&& shard: n._shards) {
+                minmax.update(shard.get_load());
             }
         } else {
             minmax.update(0);
         }
         return minmax;
     }
 
-    // Returns nullopt if capacity is not known.
-    std::optional<double> get_allocated_utilization(host_id node, const locator::load_stats& stats, uint64_t target_tablet_size) const {
+    min_max_tracker<size_t> get_shard_minmax_tablet_count(host_id node) const {
+        min_max_tracker<size_t> minmax;
+        if (_nodes.contains(node)) {
+            auto& n = _nodes.at(node);
+            for (auto&& shard: n._shards) {
+                minmax.update(shard.tablet_count);
+            }
+        } else {
+            minmax.update(0);
+        }
+        return minmax;
+    }
+
+    // Returns nullopt if node is not known.
+    std::optional<load_type> get_allocated_utilization(host_id node) const {
```

Comment:
The old semantics was to return nullopt when capacity not know. Now we throw. Why? I think we could keep it, especially that we still return optional<>.